app/src/main/rust/src/lib.rs

```diff
@@ -79,13 +78,13 @@ where
     R: Serialize,
 {
     let buffer = deref_mut_direct_bytebuffer(env, str);
-    let html = unsafe { from_utf8_unchecked(&buffer[..limit as usize]) };
-    let value = match tl::parse(html, tl::ParserOptions::default()) {
-        Ok(dom) => match f(&dom, html) {
+    let html = String::from_utf8_lossy(&buffer[..limit as usize]);
```

Comment:
I don't think we have to check it in advance since this have unnecessary overhead.
tl takes str as bytes, and use `from_utf8_lossy` internally when we request inner text of some tags.

Reply:
Done.